src/config/backend/command_line_arguments.cpp

```diff
@@ -0,0 +1,149 @@
+#include "config/backend/command_line_arguments.h"
+
+#include <map>
+
+#include <fmt/format.h>
+#include <boost/algorithm/string/classification.hpp>
+#include <boost/algorithm/string/join.hpp>
+#include <boost/algorithm/string/split.hpp>
+
+#include "silo/common/panic.h"
+#include "silo/config/util/config_exception.h"
+
+namespace silo::config {
+
+std::string CommandLineArguments::configKeyPathToString(const ConfigKeyPath& key_path) {
+   std::vector<std::string> result;
+   for (const auto& sublevel : key_path.getPath()) {
+      for (const std::string& current_string : sublevel) {
+         result.push_back(current_string);
+      }
+   }
+   return "--" + boost::join(result, "-");
+}
+
+AmbiguousConfigKeyPath CommandLineArguments::stringToConfigKeyPath(
+   const std::string& command_line_argument
+) {
+   if (command_line_argument.size() < 3 || !command_line_argument.starts_with("--")) {
+      throw silo::config::ConfigException(fmt::format(
+         "the provided option '{}' is not a valid command line option", command_line_argument
+      ));
+   }
+
+   std::vector<std::string> delimited_strings;
+   // Remove the leading dashes and split by '-'
```

Comment:
This is obvious but confusing because the split happens a couple lines further down.
```suggestion
```

Reply:
I've also reordered the code.